File: 14. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int rows = strs.size();
        if( rows == 0 ) return "";
        int mnCharStringLength = strs[0].size();
        for( int i = 1 ; i < rows ; i++ ) { 
            if( mnCharStringLength > strs[i].size() ) {
                mnCharStringLength = strs[i].size();
            }
        }
        
        string ans = "";
        bool ok = true;
        for( int i = 0 ; i < mnCharStringLength && ok ; i++ ) {
            
           
            for( int row = 1 ; row < rows && ok; row++ ) {
                if(strs[row][i] != strs[row-1][i] ) {
                    ok = false;
                }
            }
            
            if( ok ) {
                ans += strs[0][i];
            }

        }
        return ans;
    }
};

File: python/Lakshay/14. Longest Common Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        ans = ""
        sz = len(strs)
        if sz == 0:
            return ""
        if sz == 1:
            return strs[0]
        
        for i in range(len(strs[0])):
            idx = 1
            ok = True
            while idx < sz:
                if( i  >= len(strs[idx]) ):
                    ok = False
                    break
                if strs[idx][i] != strs[0][i]:
                    ok = False
                    break
                idx = idx + 1
            if ok is True:
                ans = ans + strs[0][i]
            else:
                break
                
        return ans
                    
